src/Microsoft.Health.Fhir.Shared.Api/Features/Filters/FhirRequestContextRouteDataPopulatingFilterAttribute.cs

```diff
@@ -69,14 +70,28 @@ public override void OnActionExecuting(ActionExecutingContext context)
                             return;
                         }
 
-                        switch (bundle.Type)
+                        try
                         {
-                            case Hl7.Fhir.Model.Bundle.BundleType.Batch:
-                                fhirRequestContext.AuditEventType = AuditEventSubType.Batch;
-                                break;
-                            case Hl7.Fhir.Model.Bundle.BundleType.Transaction:
-                                fhirRequestContext.AuditEventType = AuditEventSubType.Transaction;
-                                break;
+                            switch (bundle.Type)
+                            {
+                                case Hl7.Fhir.Model.Bundle.BundleType.Batch:
+                                    fhirRequestContext.AuditEventType = AuditEventSubType.Batch;
+                                    break;
+                                case Hl7.Fhir.Model.Bundle.BundleType.Transaction:
+                                    fhirRequestContext.AuditEventType = AuditEventSubType.Transaction;
+                                    break;
+                            }
+                        }
+                        catch (InvalidCastException ex)
+                        {
+                            // If the bundle-type provided is not a valid enum value, the HL7 FHIR Model will throw an InvalidCastException,
+                            // and then we will log the audit event type as a 'bundle-invalid-type'.
+                            fhirRequestContext.AuditEventType = AuditEventSubType.BundleInvalidType;
+
+                            fhirRequestContext.BundleIssues.Add(new OperationOutcomeIssue(
+                              OperationOutcomeConstants.IssueSeverity.Error,
+                              OperationOutcomeConstants.IssueType.Exception,
+                              ex.Message));
```

Comment:
Since this isn't a recoverable error I don't think this belongs here. BundleIssues are mainly used for reporting issues during Search requests

Reply:
For some reason if I throw a BadRequest, it does return a 500 error instead of 400 as we want, so I added this as a work around but open to feedback, I will ask Fernando since he has more context!